core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -970,6 +970,11 @@ protected void runTask(Task task) throws Exception {
             @Override
             protected void onInitMilestoneAttained(InitMilestone milestone) {
                 initLevel = milestone;
+                if (milestone==PLUGINS_PREPARED) {
+                    // set up Guice to enable injection as early as possible
+                    // before this milestone, ExtensionList.ensureLoaded() won't actually try to locate instances
```

Comment:
> I don't think JENKINS-22050 is a problem

I think it is; working on demonstrating it.

> I'd think this change is going to make that fix easier by allowing the reactor to reason about what initializer methods need.

I hope so. In the meantime, the force-loading of components even earlier than we already do _looks_ like a step in the wrong direction.